src/services/facilityService-pre-migration.ts

```diff
@@ -1,51 +1,88 @@
-import { Transaction } from 'firebase-admin/firestore'
 import * as gqlTypes from '../typeDefs/gqlTypes.js'
 import * as dbSchema from '../typeDefs/dbSchema.js'
 import { ErrorCode, Result } from '../result.js'
-import { dbInstance } from '../firebaseDb.js'
-import { hasSpecialCharacters, isValidEmail, isValidPhoneNumber, isValidWebsite } from '../../utils/stringUtils.js'
-import { MapDefinedFields } from '../../utils/objectUtils.js'
-import { updateHealthcareProfessionalsWithFacilityIdChanges } from './healthcareProfessionalService-pre-migration.js'
 import { logger } from '../logger.js'
-import { createAuditLog } from './auditLogService.js'
-import { chunkArray } from '../../utils/arrayUtils.js'
-import { buildBaseFacilitiesQuery } from '../../utils/searchFacilityQueryUtils.js'
-import { Timestamp } from 'firebase-admin/firestore'
+import { getSupabaseClient } from '../supabaseClient.js'
+import { createAuditLogSQL } from './auditLogServiceSupabase.js'
+import { validateIdInput, validateCreateFacilityInput, validateFacilitiesSearchInput, validateUpdateFacilityInput } from '../validation/validateFacility.js'
+import { buildFacilityUpdatePatch, applyFacilityFilters } from './helperFunctionsServices.js'
 
 /**
  * Gets the Facility from the database that matches on the id.
- * @param id A string that matches the id of the Firestore Document for the Facility.
+ * @param id A string that matches the id of the Supabase for the Facility.
  * @returns A Facility object.
  */
-export const getFacilityById = async (id: string)
-: Promise<Result<gqlTypes.Facility>> => {
+export const getFacilityById = async (
+    id: string
+): Promise<Result<gqlTypes.Facility>> => {
     try {
+        // Validate the incoming id
         const validationResult = validateIdInput(id)
 
         if (validationResult.hasErrors) {
-            logger.warn(`Validation Error: User passed in invalid id: ${id}}`)
-            return validationResult as Result<gqlTypes.Facility>
+            logger.warn(`Validation Error: User passed in invalid facility id: ${id}`)
+            return {
+                data: {} as gqlTypes.Facility,
+                hasErrors: true,
+                errors: (validationResult.errors ?? []).map(err => ({
+                    ...err,
+                    field: 'getFacilityById'
+                }))
+            }
+        }
+
+        const supabase = getSupabaseClient()
+
+        // Fetch the facility row
+        const { data: facilityRow, error: facilityRowError } = await supabase
+            .from('facilities')
+            .select('*')
+            .eq('id', id)
+            .single()
+
+        // Supabase says "no rows" and tests want INTERNAL_SERVER_ERROR, not NOT_FOUND.
+        if (facilityRowError?.code === 'PGRST116' || !facilityRow) {
```

Comment:
@Ghezz9 NIT: can you add a link to the document of PGRST116. I would also advise you do to an utility function because anyone not aware about the details of supabase would wonder what this is error is.